python/descriptor_containers.c

```diff
@@ -391,24 +395,24 @@ static PyObject *PyUpb_ByNameMap_Items(PyObject *_self, PyObject *args) {
   int n = self->funcs->base.get_elem_count(self->parent);
   PyObject *ret = PyList_New(n);
   PyObject *item;
+  PyObject *py_elem;
   if (!ret) return NULL;
   for (int i = 0; i < n; i++) {
-    item = PyTuple_New(2);
-    if (!item) goto err1;
     const void *elem = self->funcs->base.index(self->parent, i);
-    PyObject *py_elem = self->funcs->base.get_elem_wrapper(elem);
-    if (!py_elem) goto err2;
+    item = PyTuple_New(2);
+    py_elem = self->funcs->base.get_elem_wrapper(elem);
+    if (!item || !py_elem) goto error;
     PyTuple_SetItem(item, 0,
                     PyUnicode_FromString(self->funcs->get_elem_name(elem)));
     PyTuple_SetItem(item, 1, py_elem);
     PyList_SetItem(ret, i, item);
   }
   return ret;
 
-err2:
-  Py_DECREF(item);
-err1:
-  Py_DECREF(ret);
+error:
+  if (!py_elem) Py_DECREF(py_elem);
```

Comment:
`if (py_elem)` not `if (!py_elem)`

Reply:
I realized I could replace these with `Py_XDECREF()` for a more concise/foolproof NULL-check.